lib/plugins/wallet/cardano/mnemonic.js

```diff
@@ -0,0 +1,64 @@
+const fs = require('fs')
+const cardanoApi = require('./client')
+const {sha256} = require('js-sha256')
+
+const coinUtils = require('../../../coin-utils')
+const options = require('../../../options')
+
+const cryptoRec = coinUtils.getCryptoCurrency('ADA')
+const configPath = coinUtils.configPath(cryptoRec)
+
+function getCardanoMnemonic() {
+  return fs
+    .readFileSync(options.cardanoMnemonicPath, 'utf8')
+    .trim()
+    .split(/[ \r\n]+/)
+}
+
+function getCardanoMnemonicHashed() {
+  return sha256(getCardanoMnemonic().join(' '))
+}
+
+function getWalletConfiguration() {
+  return JSON.parse(fs.readFileSync(configPath))
+}
+
+function saveWalletConfiguration(configuration) {
+  fs.writeFileSync(configPath, JSON.stringify(configuration))
+}
+
+async function walletId() {
+  let configuration = getWalletConfiguration()
+
+  if (
+    !configuration.mnemonicHash ||
+    configuration.mnemonicHash !== getCardanoMnemonicHashed()
+  ) {
+    // create new wallet from mnemonic
+    const mnemonic = getCardanoMnemonic()
+
+    const walletId = await cardanoApi.newWallet(mnemonic)
+    const accountIndex = await cardanoApi.newAccount(walletId)
+
+    configuration = {
+      walletId,
+      accountIndex,
+      mnemonicHash: getCardanoMnemonicHashed()
+    }
+
+    saveWalletConfiguration(configuration)
+  }
+
+  if (!configuration.walletId || !configuration.accountIndex) {
+    throw new Error('Malformed Cardano configuration')
+  }
+
+  return Promise.resolve({
+    walletId: configuration.walletId,
+    index: configuration.accountIndex
+  })
+}
+
+module.exports = {
+  walletId
```

Comment:
I am not sure what should be the name of this module. Is `mnemonic` convention across the codebase? As it is true that this does operations on mnemonic however from the caller perspective it gives `walletId` which is the public key I suppose. And in general, this is lamassu `walletId` right? So maybe it also might make sense to call this module `lamassuWallet` although I am not really sure (and maybe I missed something). What do you think?

Reply:
`walletId` is something like a public key, but it isn't for Lamassu server, but for Cardano SL - we need this to work with our wallet over API - so maybe `cardanoWallet`? Because as you pointed, mnemonic "layer" is there only as support for wallet layer, so `wallet` in the name makes me sense.